BrickKit/bricks/src/main/java/com/wayfair/brickkit/viewholder/factory/BrickViewHolderFactory.java

```diff
@@ -0,0 +1,114 @@
+            // for avoiding app crashes.
+            if (BaseBrick.DEFAULT_LAYOUT_RES_ID == data.getViewType()) {
+                // By creating and using an empty brick view holder, the view holder can be in
+                // place in scenarios.  i.e. when bricks are null and a view holder cannot be
+                // created.
+                viewHolder = createEmptyBrickViewHolder(context);
+            } else {
+                // Since the view type (layout id) is set, create a standard brick view holder.
+                viewHolder = createViewHolderWithViewType(
+                        data.getParent(),
+                        data.getViewType(),
+                        data.getBrickProvider());
+            }
+        } catch (AssertionError ae) {
+            // This shouldn't happen
+            Log.wtf(data.getLogTag(), "Unable to get the layout inflater. ");
+        } catch (Resources.NotFoundException nfe) {
+            Log.w(data.getLogTag(), "Unable to find the resource. ", nfe);
+        } catch (NullPointerException npe) {
+            Log.w(data.getLogTag(), "A brick or the context is null and shouldn't be.");
+        }
+
+        // Since the view holder could be null, if it is, create an empty version and return
+        // its reference instead of returning a null object reference.
+        return null == viewHolder ? createEmptyBrickViewHolder(context) : viewHolder;
+    }
+
+    /**
+     * Creates an {@link com.wayfair.brickkit.viewholder.EmptyBrickViewHolder }
+     *
+     * @param context used to create the {@link EmptyBrickView}
+     * @return the newly created EmptyBrickViewHolder
+     */
+    /* private package */
+    BrickViewHolder createEmptyBrickViewHolder(@NonNull Context context) {
+        EmptyBrickView emptyBrickView = new EmptyBrickView(context);
+        return new EmptyBrickViewHolder(emptyBrickView);
+    }
+
+    /**
+     * Creates a {@link BrickViewHolder} based on the {@param #viewType}.
+     *
+     * @param parent   of the view, passed into the {@link BrickViewHolder}'s constructor.
+     * @param viewType of the view to create.  It can be a @LayoutRes id.
+     * @param provider provides bricks, based on resource id
+     * @return a newly created {@link BrickViewHolder}
+     * @throws AssertionError              thrown when the layout inflater can't be created
+     * @throws Resources.NotFoundException when the view type is zero or the resource cannot be located
+     * @throws NullPointerException        in cases, such as brick creation. (Just a fail safe)
+     */
+    /* private package */
+    BrickViewHolder createViewHolderWithViewType(@NonNull ViewGroup parent,
+                                                 int viewType,
+                                                 @NonNull BrickProvider provider)
+            throws AssertionError, Resources.NotFoundException, NullPointerException {
+
+        Context context = parent.getContext();
```

Comment:
should probably validate viewType is not 0 or -1

Reply:
I forget offhand why it can be a -1.  I'll have to look into that one.

Reply:
Not here.  If it is, the Resource.NotFoundException is thrown and caught in the calling method, above.